File: WEEK 2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int main(void)
{
    string test_string = get_string("Text: ");
    int L = 0, W = 0, S = 0;
    for (int n = 0; n <= strlen(test_string); n++)
    {
        if (test_string[n] == ' ' || test_string[n] == '\0')
        {
            W++;
        }
        else if (test_string[n] == '.' || test_string[n] == '?' || test_string[n] == '!')
        {
            S++;
        }
        else if (isalpha(test_string[n]))
        {
            L++;
        }
    }

    const float AVERAGE = 100.0 / W;
    float CL_index = (0.0588 * AVERAGE * L) - (0.296 * AVERAGE * S) - 15.8;

    if (CL_index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (CL_index > 16)
    {
        printf("Grade 16+\n");
    }
    else{
        printf("Grade %i\n",(int) round(CL_index));
    }

}

File: WEEK 6/sentimental-readability/readability.py
from cs50 import get_string

# Prompt user for input
text = get_string("Text: ")

# Compute the number of letters, words and sentences in the text
num_letters = 0
num_words = 1
num_sentences = 0
for ch in text:
    if ch.isalpha():
        num_letters += 1
    elif ch == " ":
        num_words += 1
    elif ch in [".", "!", "?"]:
        num_sentences += 1

# Compute the Coleman-Liau index
L = num_letters * 100 / num_words
S = num_sentences * 100 / num_words
index = 0.0588 * L - 0.296 * S - 15.8

# Print the grade level
if index < 1:
    print("Before Grade 1")
elif index >= 16:
    print("Grade 16+")
else:
    print(f"Grade {round(index)}")